src/handlers/grep.rs

```diff
@@ -900,4 +900,46 @@ mod tests {
         let apparently_grep_output = "src/co-7-fig.rs:xxx";
         assert_eq!(parse_grep_line(apparently_grep_output), None);
     }
+
+    #[test]
+    fn test_get_code_style_sections() {
+        use crate::style::Style;
+        use crate::handlers::grep::get_code_style_sections;
+        use crate::paint::StyleSectionSpecifier;
+        use crate::ansi::strip_ansi_codes;
+
+        let fake_parent_grep_command = "git --doesnt-matter grep --nor-this nor_this -- nor_this";
+        let _args = FakeParentArgs::for_scope(fake_parent_grep_command);
+
+        let miss = Style::new();
+        let hit = Style {
+            is_emph: true,
+            ..miss
+        };
+
+        let escape = "\x1B";
+        let working_example = format!("foo/bar/baz.yaml{escape}[36m:{escape}[m1090{escape}[36m:{escape}[m  - {escape}[1;31mkind: Service{escape}[mAccount");
+        let stripped = strip_ansi_codes(&working_example);
+        let grep = parse_grep_line(&stripped).unwrap();
+
+        assert_eq!(
+            get_code_style_sections(&working_example, hit, miss, &grep),
+            Some(StyleSectionSpecifier::StyleSections(vec![
+                                                      (miss, "  - "),
+                                                      (hit, "kind: Service"),
+                                                      (miss, "Account")
+            ]))
+        );
+
+        let broken_example = format!("foo/bar/baz.yaml{escape}[36m:{escape}[m2{escape}[36m:{escape}[m{escape}[1;31mkind: Service{escape}[m");
```

Comment:
These variable-names aren't really applicable anymore. `working_example` was the one that worked before the changes were made to `get_code_style_sections`, `broken_example` was the one that didn't. That wouldn't actually be a useful distinction going forward; maybe we should rename them?